extensions/markdown-language-features/src/languageFeatures/documentLinkProvider.ts

```diff
@@ -257,33 +270,33 @@ export class MdLinkProvider implements vscode.DocumentLinkProvider {
 	}
 
 	public async getAllLinks(document: SkinnyTextDocument): Promise<MdLink[]> {
+		const codeInDocument = await findCode(document, this.engine);
 		return Array.from([
-			...(await this.getInlineLinks(document)),
-			...this.getReferenceLinks(document),
-			...this.getLinkDefinitions(document),
-			...this.getAutoLinks(document),
+			...this.getInlineLinks(document, codeInDocument),
+			...this.getReferenceLinks(document, codeInDocument),
+			...this.getLinkDefinitions(document, codeInDocument),
+			...this.getAutoLinks(document, codeInDocument),
 		]);
 	}
 
-	private async getInlineLinks(document: SkinnyTextDocument): Promise<MdLink[]> {
+	private *getInlineLinks(document: SkinnyTextDocument, codeInDocument: CodeInDocument): Iterable<MdLink> {
 		const text = document.getText();
 
 		const results: MdLink[] = [];
-		const codeInDocument = await findCode(document, this.engine);
 		for (const match of text.matchAll(linkPattern)) {
 			const matchImageData = match[4] && extractDocumentLink(document, match[3].length + 1, match[4], match.index);
-			if (matchImageData && !isLinkInsideCode(codeInDocument, matchImageData)) {
-				results.push(matchImageData);
+			if (matchImageData && !isLinkInsideCode(codeInDocument, matchImageData.source.hrefRange)) {
+				yield matchImageData;
 			}
 			const matchLinkData = extractDocumentLink(document, match[1].length, match[5], match.index);
-			if (matchLinkData && !isLinkInsideCode(codeInDocument, matchLinkData)) {
-				results.push(matchLinkData);
+			if (matchLinkData && !isLinkInsideCode(codeInDocument, matchLinkData.source.hrefRange)) {
+				yield matchLinkData;
 			}
 		}
 		return results;
```

Comment:
Since this is now a generator, you should remove the `return` statement (and the `results` variable entirely)